src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+
+export interface AccordionProps {
+    type?: 'base' | 'single' | undefined
+    data: AccordionItemProps[]
+}
+
+export interface AccordionItemProps {
+    id: string
+    title: string
+    content: string
+}
+
+const Accordion: React.FC<AccordionProps> = ({ type, data }): JSX.Element => {
+    const types = type === undefined ? 'base' : type
+    const accordionClass = types === 'single' ? 'accordion__input--single' : ''
+    const accordionType = types === 'single' ? 'radio' : 'checkbox'
+    const accordionId = types === 'single' ? 's' : ''
+    const [ariaState, setAriaState] = useState(
+        new Array(data.length).fill(true)
+    )
+    const [checkedState, setCheckedState] = useState(
+        new Array(data.length).fill(false)
+    )
+    const [divState, setDivState] = useState(new Array(data.length).fill(false))
+    const toggleHandler = (position: number) => {
+        
+        const updatedAriaState = ariaState.map((item, index) =>
+            index === position ? !item : item
+        )
+        setAriaState(updatedAriaState)
+    }
+    const handleOnChange = (
+        position: number,
+        accordionTypes: 'base' | 'single'
+    ) => {
+        if (accordionTypes === 'base') {
+            const updatedCheckedState = checkedState.map((item, index) =>
+                index === position ? !item : item
+            )
+            setCheckedState(updatedCheckedState)
+        } else {
+            const updatedCheckedState = checkedState.map((item, index) =>
+                index === position ? !item : false
+            )
+            setCheckedState(updatedCheckedState)
+            const updatedDivState = divState.map((item, index) => {
+                if (index === position && checkedState[index] === false) {
+                    return true
+                }
+                return false
+            })
+            setDivState(updatedDivState)
+        }
+    }
+    return (
+        <>
+            {data.map((accordion, index) => (
+                <div key={index} className='c-accordion'>
+                    <input
+                        className={`accordion__input ${accordionClass}`}
+                        id={`accordion-id-${index}${accordionId}`}
+                        name={`accordion-${accordionType}`}
+                        type='checkbox'
+                        aria-expanded={!ariaState[index]}
```

Comment:
Good job on the base variant. 
Aria is still glitching for the single variant

Reply:
Done, fixed